### PR TITLE
feat(feature-toggles): Add feature object to standardize feature toggles

### PR DESCRIPTION
**Pull request by JChan106**
Purpose is to enhance the Preview SDK to standardize the usage of feature toggles. This takes in a features object passed in from the options object and sets to the Preview options field. This allows Preview items to have access to the features passed in from the Preview SDK caller.

### CODE REVIEW

**Review by jstoffan — CHANGES_REQUESTED** [review]
We should avoid introducing global variables into the SDK. Let's discuss the approach.

File: src/lib/Preview.js [hunk 1]
```diff
@@ -1015,6 +1015,11 @@ class Preview extends EventEmitter {
                 this.enableViewers(viewerName);
             }
         });
+
+        // Features
+        // This makes features available everywhere that features is passed in, which includes
+        // all of the viewers for different files
+        this.options.features = options.features;
```
Comment (jstoffan) [inline src/lib/Preview.js:1022]: Can we move this before the enable/disable viewer logic? Should it have a default value, such as an empty object?

File: src/lib/__tests__/Preview-test.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ let preview;
 let containerEl;
 
 describe('lib/Preview', () => {
+    const features = { shouldUseBatman: { enabled: true } };
```
Comment (jstoffan) [inline src/lib/__tests__/Preview-test.js:41]: Probably best to avoid copyrighted characters. 😄

File: src/lib/__tests__/featureChecking-test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { getFeatureConfig, isFeatureEnabled } from '../featureChecking';
```
Comment (jstoffan) [inline src/lib/__tests__/featureChecking-test.js:1]: Can this be converted to a TS file, as well?

File: src/lib/__tests__/featureChecking-test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+const Box = {
```
Comment (jstoffan) [inline src/lib/__tests__/featureChecking-test.js:3]: Can we remove the `Box` wrapper?

File: src/lib/featureChecking.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+const isFeatureEnabled = (features: FeatureConfig, featureName: string): boolean => {
+    return !!get(features, featureName, false);
+};
+
+const getFeatureConfig = (features: FeatureConfig, featureName: string): FeatureConfig => {
+    return get(features, featureName, {});
+};
+
+export { isFeatureEnabled, getFeatureConfig };
```
Comment (jstoffan) [inline src/lib/featureChecking.ts:19]: Can we export inline?

```
export const getFeatureConfig ...
```

File: src/lib/Preview.js (hunk 1 above)
Comment (JChan106) [reply src/lib/Preview.js:1022]: Good point, will do

File: src/lib/__tests__/featureChecking-test.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+const features = {
+    shouldUseCar: {
+        enabled: true,
+    },
+    shouldUsePlane: {
+        enabled: false,
+    },
+};
+
+describe('isFeatureEnabled', () => {
```
Comment (jstoffan) [inline src/lib/__tests__/featureChecking-test.ts:12]: Minor, but we typically maintain one top-level `describe` per test file. Can we wrap these two?

**Review by jstoffan — APPROVED** [review]

File: src/lib/constants.js [hunk 7]
```diff
@@ -102,7 +102,7 @@ export const PDFJS_WIDTH_PADDING_PX = 40; // Should match SCROLLBAR_PADDING in p
 export const PDFJS_HEIGHT_PADDING_PX = 5; // Should match VERTICAL_PADDING in pdf_viewer.js
 
 export const DOC_LEGACY_STATIC_ASSETS_VERSION = '2.76.0'; // IE11 only
-export const DOC_STATIC_ASSETS_VERSION = '2.93.0'; // Modern browsers
+export const DOC_STATIC_ASSETS_VERSION = '2.90.0'; // Modern browsers
```
Comment (jstoffan) [inline src/lib/constants.js:105]: Can we make this change as a separate PR?

Comment (JChan106) [reply src/lib/constants.js:105]: https://github.com/box/box-content-preview/pull/1478

**Review by jstoffan — APPROVED** [review]
